### ISSUE TITLE
improve error message on set input to `wandb.log`

### ISSUE THREAD

**Issue by yaroslavvb** [wandb/client#469]
Giving `{'value', val}` instead of `{value: val}` to `wandb.log` (by mistake) gives a hard to parse error message, `AttributeError: 'set' object has no attribute 'keys'`, this user error could be caught higher

```
Traceback (most recent call last):
  File "/home/ec2-user/anaconda3/envs/pytorch_p36/lib/python3.6/pdb.py", line 1667, in main
    pdb._runscript(mainpyfile)
  File "/home/ec2-user/anaconda3/envs/pytorch_p36/lib/python3.6/pdb.py", line 1548, in _runscript
    self.run(statement)
  File "/home/ec2-user/anaconda3/envs/pytorch_p36/lib/python3.6/bdb.py", line 434, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/home/ec2-user/train.py", line 1, in <module>
    import argparse
  File "/home/ec2-user/train.py", line 297, in main
    else:
  File "/home/ec2-user/train.py", line 273, in worker
    wandb.init(project='pytorch-curv', name=args.run_name)
  File "/home/ec2-user/train.py", line 264, in train
    import pdb; pdb.set_trace()
  File "/home/ec2-user/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/wandb/__init__.py", line 526, in log
    if any(not isinstance(key, six.string_types) for key in row.keys()):
```

**Comment by ariG23498** [comment]
Hey @yaroslavvb 
I think this issue is resolved now
![image](https://user-images.githubusercontent.com/36856589/100824535-56b19600-347c-11eb-8b24-c99216a56116.png)
